core/array.rbs

```diff
@@ -850,7 +850,7 @@ class Array[unchecked out Elem] < Object
   #
   # Array#slice is an alias for Array#[].
   #
-  def []: (int index) -> Elem
+  def []: (int index) -> Elem?
```

Comment:
This is intentional, following the type of array in TypeScript, which I regret for years. Could you revert?

I'm open to make it strict, but don't think we can do it right now because it will break many typed Ruby code. (Maybe next major release?)

Reply:
Sure, reverted :). Thanks for the info on why it's intentional.